### PR TITLE
Implement all standard functions listed in GatewayInterface

### PR DESCRIPTION
**Pull request by jtgrimes**
I found myself needing to add void() and capture() functions so I could test my project, and then it sort of took off. \Omnipay\Common\GatewayInterface lists functions which should be standard in any gateway, so I just implemented a dummy version of all of them.

### CODE REVIEW

**Comment by barryvdh** [comment]
Makes sense to me. If we would really want to complete, we should also allow for a redirect response probably though.

**Comment by delatbabel** [comment]
Noted, I have a massive backlog of real world work at the moment, will try to clear my github/omnipay backlog early next week.

**Comment by coatesap** [comment]
+1 Having the refund method on Dummy will be super-useful!

File: src/Gateway.php [hunk 1]
```diff
@@ -69,21 +77,93 @@ public function getDefaultParameters()
...
+     * @return \Omnipay\Common\Message\ResponseInterface
+     */
+    public function completePurchase(array $parameters = array())
+    {
+        return $this->createRequest('\Omnipay\Dummy\Message\TransactionReferenceRequest', $parameters);
+    }
+
+    /**
+     * @param array $parameters
+     * @return \Omnipay\Common\Message\ResponseInterface
```
Comment (coatesap) [inline src/Gateway.php:127]: It looks like these docblocks are incorrect - as far as I can tell all the public methods in Gateway always return an AbstractRequest, not a ResponseInterface...

**Comment by delatbabel** [comment]
@coatesap comment about the docblocks is correct -- these need to be fixed.

Also I see that some parts of the markup in the class docblock have changed, e.g. the headings removed.  I see no good reason for that.

File: src/Message/CardReferenceRequest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?php namespace Omnipay\Dummy\Message;
```
Comment (rquadling) [inline src/Message/CardReferenceRequest.php:1]: Is that a BOM I see before me?

Comment (delatbabel) [reply src/Message/CardReferenceRequest.php:1]: Were you expecting one? :)

Comment (rquadling) [reply src/Message/CardReferenceRequest.php:1]: Ha. It's gone now. The '<' was in red.

**Comment by delatbabel** [comment]
OK notwithstanding the fact that travis is passing the tests, this PR needs the docblocks fixed before I can merge.

**Comment by kynx** [comment]
@delatbabel The [GatewayInterface](https://github.com/thephpleague/omnipay-common/blob/77863b86c5b17bc37432537e3b8cd905cc4b8114/src/Common/GatewayInterface.php#L16) says the methods return a `RequestInterface`.

@jtgrimes I'd really like to see this merged. You OK with me doing a new PR based on this with the docblocks fixed?

**Comment by jtgrimes** [comment]
@kynx - you bet! I'm sorry I lost track of this PR - if you want to pick it up, that's great!

**Comment by rquadling** [comment]
@jtgrimes Is your fork at https://github.com/jtgrimes/omnipay-dummy now redundant? Can we move back to using https://github.com/thephpleague/omnipay-dummy? (I think so, just would like to confirm).

**Comment by jtgrimes** [comment]
@rquadling yep - it was merged in and my fork is actually behind the current repo - definitely switch back.

**Pull request merged**